### PR TITLE
Work to unify threading concepts between Workspace and ProjectSystemProjectFactory (part2)

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/67522

Continue to move existing ProjectSystemProjectFactory code over to using transformation-helper supplied by workspace.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs [hunk 1]
```diff
@@ -78,57 +77,76 @@ public ProjectSystemProjectFactory(Workspace workspace, IFileChangeWatcher fileC
...
-                            SolutionPath,
-                            projects: new[] { projectInfo },
-                            analyzerReferences: w.CurrentSolution.AnalyzerReferences)
-                        .WithTelemetryId(SolutionTelemetryId));
-                }
-                else
-                {
-                    w.OnProjectAdded(projectInfo);
-                }
+                await w.SetCurrentSolutionAsync(
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs:115]: this now uses the new helper, spinning until it 'wins' and actually makes the change to the workspace.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski ptal.  fairly simple change.

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs [hunk 2]
```diff
@@ -78,57 +77,76 @@ public ProjectSystemProjectFactory(Workspace workspace, IFileChangeWatcher fileC
 
         public async Task<ProjectSystemProject> CreateAndAddToWorkspaceAsync(string projectSystemName, string language, ProjectSystemProjectCreationInfo creationInfo, ProjectSystemHostInfo hostInfo)
         {
-            var id = ProjectId.CreateNewId(projectSystemName);
+            var projectId = ProjectId.CreateNewId(projectSystemName);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs:80]: just a rename as 'id' was very confusing to me.

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs [hunk 3]
```diff
@@ -78,57 +77,76 @@ public ProjectSystemProjectFactory(Workspace workspace, IFileChangeWatcher fileC
...
                 hostInfo,
-                id,
+                projectId,
                 displayName: projectSystemName,
                 language,
-                assemblyName: assemblyName,
+                assemblyName,
+                creationInfo.CompilationOptions,
+                creationInfo.FilePath,
+                creationInfo.ParseOptions);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs:93]: just removed named options i thought were redundant.

File: src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs [hunk 4]
```diff
@@ -78,57 +77,76 @@ public ProjectSystemProjectFactory(Workspace workspace, IFileChangeWatcher fileC
...
+                    name: projectSystemName,
+                    assemblyName,
+                    language,
+                    compilationOutputFilePaths: default, // will be updated when command line is set
+                    SourceHashAlgorithms.Default, // will be updated when command line is set
+                    filePath: creationInfo.FilePath,
+                    telemetryId: creationInfo.TelemetryId),
                 compilationOptions: creationInfo.CompilationOptions,
-                filePath: creationInfo.FilePath,
                 parseOptions: creationInfo.ParseOptions);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs:111]: pulled creation of this info singleton up out of hte lambda.

**Comment by CyrusNajmabadi** [comment]
@dibarbet  ptal as jason is away currently.  thanks!

**Review by dibarbet — APPROVED** [review]

**Pull request merged**